src/v2/Components/FluidGrid.tsx

```diff
@@ -0,0 +1,60 @@
+import React from "react"
+import styled from "styled-components"
+import { Box, BoxProps, CSSGrid } from "@artsy/palette"
+import {
+  GridColumnProps,
+  ResponsiveValue,
+  gridColumn,
+  system,
+} from "styled-system"
+
+/**
+ * A 12-column fluid grid. Collapses to a single column at mobile breakpoints.
+ */
+export const Grid = styled(CSSGrid)``
+
+Grid.defaultProps = {
+  gridColumnGap: 2,
+  gridRowGap: 2,
+  gridTemplateColumns: "repeat(12, 1fr)",
+}
+
+const colSpan = system({
+  span: {
+    property: "gridColumn",
+    transform: value => value && `span ${value}`,
+  },
+})
+
+type ColSpanProps = { span?: ResponsiveValue<number> } & GridColumnProps
+
+const __Col__ = styled(Box)<ColSpanProps>`
+  ${colSpan}
+  ${gridColumn}
+  outline: 1px dotted red;
+`
+
+export const Col: React.FC<BoxProps & ColSpanProps> = ({ span, ...rest }) => {
+  if (typeof span === "number") {
+    return (
+      <__Col__
+        span={[undefined, span]}
+        gridColumn={["1 / -1", undefined]}
+        {...rest}
+      />
+    )
+  }
+
+  return <__Col__ span={span} {...rest} />
```

Comment:
This indirection is a bit weird but basically what I wanted was to simplify the notion of colspan by accepting a number and transforming it to `span N`, however, by default forcing `gridColumn="1 / -1"` for mobile breakpoints which will span the full row.

Reply:
I think I can simplify this a bit and accept valid `gridColumn` values in the `span` shortcut.